### PR TITLE
Create health check for SyncService

### PR DESCRIPTION
**Pull request by cyberbono3**
Created health check for SyncService 

|Resolves] #1206

---
1. Added health check to Status() function in beacon-chain/sync/service.go
2. Added health check for unsynced service in run() in service.go
3. Added unit test service_test.go for Status()

### CODE REVIEW

File: beacon-chain/sync/service.go [hunk 1]
```diff
@@ -80,13 +81,18 @@ func (ss *Service) Stop() error {
...
+		return ss.failStatus
+	}
 	return nil
 }
 
 func (ss *Service) run() {
 	ss.Querier.Start()
 	synced, err := ss.Querier.IsSynced()
 	if err != nil {
+		ss.failStatus = err
```
Comment (prestonvanloon) [inline beacon-chain/sync/service.go]: The server crashes on the next line so there is no point in setting this.

File: beacon-chain/sync/service.go [hunk 2]
```diff
@@ -80,13 +81,18 @@ func (ss *Service) Stop() error {
 // Status always returns nil.
 // TODO(1206): Add service health checks.
 func (ss *Service) Status() error {
+	if ss.failStatus != nil {
```
Comment (prestonvanloon) [inline beacon-chain/sync/service.go]: Is this the only check you are going to add?

What about when the client is not synced? I think that is an unhealthy condition

File: beacon-chain/sync/service.go (hunk 1 above)
Comment (cyberbono3) [reply beacon-chain/sync/service.go]: removed this line

File: beacon-chain/sync/service.go (hunk 2 above)
Comment (cyberbono3) [reply beacon-chain/sync/service.go]: Added TestStatus() and unhealthy status for unsynced service!

File: WORKSPACE [hunk 3]
```diff
@@ -465,7 +465,7 @@ go_repository(
 
 go_repository(
```
Comment (terenc3t) [inline WORKSPACE]: These should not be part of this PR

File: shared/iputils/external_ip_test.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package iputils
```
Comment (terenc3t) [inline shared/iputils/external_ip_test.go]: This one too

File: shared/p2p/service.go [hunk 5]
```diff
@@ -240,7 +240,18 @@ func (s *Server) Send(msg proto.Message, peer Peer) {
 	s.Broadcast(msg)
 }
```
Comment (terenc3t) [inline shared/p2p/service.go]: This one too

File: WORKSPACE (hunk 3 above)
Comment (prestonvanloon) [reply WORKSPACE]: +1 Revert changes to WORKSPACE file

File: beacon-chain/sync/service.go [hunk 6]
```diff
@@ -93,6 +99,8 @@ func (ss *Service) run() {
 	if synced {
 		ss.RegularSync.Start()
 		return
+	} else {
+		ss.failStatus = errors.New("Service is unhealthy")
```
Comment (prestonvanloon) [inline beacon-chain/sync/service.go]: Please do not use a failure status for something you can query. 

Your Status method should look something like this:
```go
func (ss *SyncServer) Status() error {
  synced, err := ss.Querier.IsSynced() 
  if err != nil {
    return err
   }
  if !synced {
    return errors.New("not in sync")
  }
}
```

**Comment by cyberbono3** [comment]
Review required

**Comment by cyberbono3** [comment]
@terenc3t   Yeah you are right something went wrong , when i started building....should I delete this PR and resubmit correct one?

**Comment by terenc3t** [comment]
I think you just need to rebase with the master
